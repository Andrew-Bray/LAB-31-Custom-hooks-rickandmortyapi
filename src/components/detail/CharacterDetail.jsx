import React from 'react';
import PropTypes from 'prop-types';
import styles from './characterdetail.css';
import { useTheme } from '../../state/siteTheme';


const CharacterDetail = ({ name, image, status, species, origin }) => {
  const { theme } = useTheme();
  return (
    <figure 
      data-testid="one-character"
      className={`${styles.CharacterDetail} ${styles[theme]}`}>
      <h2>{name}</h2>
      <img src={image} alt={name} />
      <div className={styles.DetailBox}>
        <span>Species: {species}</span>
        <span>Status: {status}</span>
      </div>
      <h3>Origin: {origin}</h3>
    </figure>
  );
};

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  status: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
};


export default CharacterDetail;
