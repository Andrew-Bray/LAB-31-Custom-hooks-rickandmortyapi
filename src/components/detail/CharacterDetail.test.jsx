import React from 'react';
import { render } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';
import ThemeProvider from '../../state/siteTheme';

describe('CharacterDetail component', () => {
  it('renders CharacterDetail', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <CharacterDetail />
      </ThemeProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
