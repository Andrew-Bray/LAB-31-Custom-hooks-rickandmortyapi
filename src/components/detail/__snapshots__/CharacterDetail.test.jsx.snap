// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <figure
    class="CharacterDetail Puny"
    data-testid="one-character"
  >
    <h2 />
    <img />
    <div
      class="DetailBox"
    >
      <span>
        Species: 
      </span>
      <span>
        Status: 
      </span>
    </div>
    <h3>
      Origin: 
    </h3>
  </figure>
</DocumentFragment>
`;
